### PR TITLE
EU-15-add-remove-and-patch-to-eu-register

### PR DESCRIPTION
**Pull request by jenjou**

### CODE REVIEW

File: encode_utils/MetaDataRegistration/eu_register.py [hunk 1]
```diff
@@ -150,15 +159,33 @@ def main():
...
+            record_id = payload.get(RECORD_ID_FIELD, False)
+            if not record_id:
+                raise Exception(
+                    "Can't patch payload {} since there isn't a '{}' field indicating an identifier for the record to be PATCHED.".format(
+                        euu.print_format_dict(payload), RECORD_ID_FIELD))
+            payload.pop(RECORD_ID_FIELD)
+            payload.update({conn.ENCID_KEY: record_id})
+            conn.remove_and_patch(props=props_to_remove, patch=payload, extend_array_values=not overwrite_array_values)
+        elif props_to_remove and not rmpatch:
+            raise Exception("Properties to remove were specified, but --rm-patch flag was not set.")
```
Comment (p-sud) [inline encode_utils/MetaDataRegistration/eu_register.py]: Similar to an above comment. This code should be moved up to right after the `parser.parse_args()`, and use parser.error:
```suggestion
        if args. props_to_remove and not args.rmpatch:
            parser.error("Properties to remove were specified, but --rm-patch flag was not set.")
```

File: encode_utils/MetaDataRegistration/eu_register.py [hunk 2]
```diff
@@ -338,7 +365,7 @@ def create_payloads_from_tsv(profile_id, infile):
                         val = val[:-1]
                     val = [x.strip() for x in val.split(",")]
                     # Type cast tokens if need be, i.e. to integers:
-                    val = [typecast(field_name=field, value=x, data_type=item_val_type, line_num=line_count) for x in val]
+                    val = [typecast(field_name=field, value=x, data_type=item_val_type, line_num=line_count) for x in val if x != '']
```
Comment (p-sud) [inline encode_utils/MetaDataRegistration/eu_register.py]: Is there a good reason to explicitly check against the empty string? If not you could just do `if x`:
```suggestion
                    val = [typecast(field_name=field, value=x, data_type=item_val_type, line_num=line_count) for x in val if x]
```

Comment (jenjou) [reply encode_utils/MetaDataRegistration/eu_register.py]: Removed empty string check

File: encode_utils/MetaDataRegistration/eu_register.py [hunk 3]
```diff
@@ -150,15 +159,33 @@ def main():
...
             conn.post(payload, require_aliases=not no_aliases)
-        else:
+        elif rmpatch:
+            if not props_to_remove:
+                raise Exception("No properties to remove were specified. Use --patch if only patching is needed.")
+            else:
+                props_to_remove = props_to_remove.split(',')
+            record_id = payload.get(RECORD_ID_FIELD, False)
+            if not record_id:
+                raise Exception(
```
[the remark this answers is not in the document]
Comment (jenjou) [reply encode_utils/MetaDataRegistration/eu_register.py]: Changed this as well as corresponding line in the "patch" case

**Comment by jenjou** [comment]
It doesn't really make sense to specify both, but I think as the current behavior would happen to end up as intended anyway. If both --patch and --rmpatch are specified but there's no -r, it would raise the "No properties to remove" error, and if there is -r, it would default to removing and patching. But it seems like to be more explicit about expected arguments, it'd be better to make them mutually exclusive anyway?

**Comment by p-sud** [comment]
> But it seems like to be more explicit about expected arguments, it'd be better to make them mutually exclusive anyway?

I think so too.

**Comment by p-sud** [comment]
Sorry I didn't catch the split in the loop during code review! Still looks good.

**Pull request merged**